services/ui-src/src/components/modals/AddEditReportModal.tsx

```diff
@@ -84,7 +91,22 @@ export const AddEditReportModal = ({
 
   // MCPAR report payload
   const prepareMcparPayload = (formData: any) => {
-    const programName = formData["programName"];
+    const newOrExistingProgram = formData["newOrExistingProgram"];
+    const existingProgramNameSelection = formData[
+      "existingProgramNameSelection"
+    ] || {
+      label: dropdownDefaultOptionText,
+      value: "",
+    };
+    const existingProgramNameSuggestion =
+      formData["existingProgramNameSuggestion"] || "";
+    const newProgramName = formData["newProgramName"] || "";
```

Comment:
if the flag is false, will these error on submit? Do we need optional `?.` chaining?
`key "newOrExistingProgram" does not exist` etc.